generator/tree/nodes/node.py

```diff
@@ -179,6 +180,21 @@ def root(self):
             result = result._parent
         return result
 
+    def extract_subtree(self):
+        """
+        Extract the subtree of node (some nodes are copied)
```

Comment:
I am not sure I like "some"... Basically, the parent nodes are copied and the children aren't...